WIP: allow CI deployments on Azure

wesjdj opened the pull request:
/AzureDeploy

lorenzo-cavazzi commented:
Generic comment: Why make a separate workflow? A bunch of jobs overlap with the existing `pull-request-test` workflow. We can adapt it to have an additional `deploy-pr-azure` job

lorenzo-cavazzi reviewed (commented):
Generic comment: Why make a separate workflow? A bunch of jobs overlap with the existing `pull-request-test` workflow. We can adapt it to have an additional `deploy-pr-azure` job

olevski commented on .github/workflows/pull-request-test-azure.yml:
```diff
@@ -0,0 +1,184 @@
+    runs-on: ubuntu-24.04
+    steps:
+      - name: renku teardown
+        uses: SwissDataScienceCenter/renku-actions/cleanup-renku-ci-deployments@v1.15.1
```
Everything else is on 1.18.0 is there a reason why this isnt?

```suggestion
        uses: SwissDataScienceCenter/renku-actions/cleanup-renku-ci-deployments@v1.18.0
```

wesjdj replied:
Thanks, that had been changed since I created the first branch

olevski reviewed (changes requested):
One small suggestion. And I think Lorenzo did the review at the same time as me.

lorenzo-cavazzi reviewed (dismissed):
Looks good already :clap: 
I added a couple of nitpick comments, feel free to ignore them.

I added the deploy string and something seems off with `#legacy`, the tests didn't run

wesjdj commented:
> Looks good already 👏 I added a couple of nitpick comments, feel free to ignore them.
> 
> I added the deploy string and something seems off with `#legacy`, the tests didn't run

There is an issue on the cluster with csi-rclone, hence the tests not working

wesjdj commented on .github/workflows/pull-request-test.yml:
```diff
@@ -53,29 +53,35 @@ jobs:
+      - id: deploy-comment-azure
+        uses: SwissDataScienceCenter/renku-actions/check-pr-description@v1.18.0
+        with:
+          string: /AzureDeploy
```
I think I'll leave as is for now. If/when we fully move to the Azure dev cluster, we will get /deploy to deploy to Azure instead.

lorenzo-cavazzi commented:
> > Looks good already 👏 I added a couple of nitpick comments, feel free to ignore them.
> > I added the deploy string and something seems off with `#legacy`, the tests didn't run
> 
> There is an issue on the cluster with csi-rclone, hence some sessions are not able to start

Shouldn't the check still show up? I'm surprised to see that the job was skipped even with the #legacy option in the deploy string.

![image](https://github.com/user-attachments/assets/ee20b6dd-3f78-4340-9dd3-57b50924a5a2)

Also, this hasn't been addressed https://github.com/SwissDataScienceCenter/renku/pull/3981#discussion_r2046206733
Let's be consistent in the version we use for renku-actions.
